src/common/2d/v_2ddrawer.h

```diff
@@ -123,20 +125,21 @@ class F2DDrawer
 		bool useTransform;
 		DMatrix3x3 transform;
 
-		DShape2D* shape2D;
+		std::shared_ptr<DShape2DBufferInfo> shape2DBufInfo;
 		int shape2DBufIndex;
 		int shape2DIndexCount;
 		int shape2DCommandCounter;
 
 		RenderCommand()
 		{
 			memset(this, 0, sizeof(*this));
+			shape2DBufInfo = nullptr;
```

Comment:
This assignment is useless, as it holds no object by default. But it’s a very minor thing comparing to `memset()` of `shared_ptr<>`.

Reply:
As far as I can tell `std::shared_ptr` has a copy constructor `constexpr shared_ptr( std::nullptr_t ) noexcept;` which means this will assign a valid `std::shared_ptr` to `shape2DBufInfo` that represents a `nullptr`. If I'm wrong about this then I can change this to `shape2DBufInfo.reset();` instead (along with the other assignment to `nullptr` in `OnDestroy`).

Reply:
I went ahead and changed this to `.reset()` anyway since it's easier to read.

Reply:
You missed the point of my first comment. It is possible that after doing `memset()` on `shared_ptr<>`  instance, it will be in corrupted state. It doesn’t matter what constructors/operators/functions you will call on it, nothing will guarantee that the object will become valid again.
Either use own reference counting pointer that is `memset()` friendly, or remove it and initialize all members explicitly.